### PR TITLE
5033: (fix) adds locationAnalyzer to default-app template

### LINKED ISSUE #5033: Missing analyzer backend if using @backstage/create-app
When bootstrapping a backstage application with `npx @backstage/create-app` the catalog option to `Register an existing component` offers the ability to import repositories without `catalog-info.yaml`.
After configuring GitHub auth and trying to import a repository like `https://github.com/organization/project`, the backend reports a 404 error.

`Failed to generate entity definitions. Received http response 404: Not Found`
image

When building from source (`backstage/backstage`) the functionality is working fine.

## Expected Behavior

The wizard should provide me with information about a PullRequest that is going to be opened with base configuration of the component.

## Current Behavior

The Website/Console/Backend report a 404 for the `http://localhost:7000/api/catalog/analyze-location` path.

## Steps to Reproduce

1. Bootstrap a new application with `npx @backstage/create-app`
2. Configure auth for Github (https://backstage.io/docs/tutorials/quickstart-app-auth)
3. Try to `Register an existing component` with a repository without component configuration

## Context

I'm trying to setup a new backstage application for my company and to import existing projects to the catalog.

## Your Environment

I've tested different kind of combinations of browsers, databases and configurations.

- NodeJS Version (v14):
- Operating System and Version (MacOS BigSur):
- Browser Information: Firefox/Chrome/Safari
- SQLite and Postgresql

### PR DESCRIPTION
**Pull request by azerbe**
Signed-off-by: Alexander Zerbe <alex@aqube.de>

## Hey, I just made a Pull Request!

Resolves #5033 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — APPROVED** [review]
Yes!

**Comment by freben** [comment]
Oh dang, forgot one thing. The changeset. You'll need to do `yarn changeset` and mark a patch release for this package. AND you'll write a little bit of text in that changeset file about how people can do the corresponding thing by hand in their already-created env. See previous changelog entries for that package for inspiration

**Comment by azerbe** [comment]
Alright. Will add this later.

**Comment by freben** [comment]
Also a minor typo build error https://github.com/backstage/backstage/pull/5037/checks?check_run_id=2155347618

File: packages/create-app/templates/default-app/packages/backend/src/plugins/catalog.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { useHotCleanup } from '@backstage/backend-common';
 import {
   CatalogBuilder,
   createRouter,
-  runPeriodically
+  runPeriodically,
+  locationAnalyzer,
```
Comment (freben) [inline packages/create-app/templates/default-app/packages/backend/src/plugins/catalog.ts:6]: This line should be on line 14-16

**Review by freben — CHANGES_REQUESTED** [review]

File: .changeset/curly-fishes-speak.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+   } = await builder.build();
+   // ...
+   return await createRouter({
+     entitiesCatalog,
+     locationsCatalog,
+     higherOrderOperation,
++    locationAnalyzer,
+     logger: env.logger,
+   });
+```
```
Comment (freben) [inline .changeset/curly-fishes-speak.md:31]: I rudely added the diff here on top of your changes, to make sure that the user sees it clearly in front of them. They won't be "in the context of" this repo necessarily when reading this; they have their own private repo and won't have the create-app source easily available in front of them.

**Review by freben — APPROVED** [review]

**Review by LEBATRUNG — APPROVED** [review]

**Pull request merged**